util/templateUtils.go

```diff
@@ -40,5 +40,13 @@ func GenerateAndWriteTemplateFile(
 	if err != nil {
 		return errors.Wrapf(err, "Failed to write values to %s file\n", fileName)
 	}
+
+	// Change file permissions to 600.
+	// This is to ensure that the file is only readable and writable by the owner.
+	// As it might contain some sensitive information.
+	err = os.Chmod(fileDirectory+fileName, 0600)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to change %s file permissions\n", fileName)
+	}
```

Comment:
There are two issues with this approach.
1. This causes the file to be temporarily readable to others. We should instead create the file with the appropriate file permissions to begin with, by calling [`os.OpenFile`](https://pkg.go.dev/os#OpenFile) instead of `os.Create`.
2. This causes all files to be created with 0600 permissions, which may not be desirable. We should instead have `GenerateAndWriteTemplateFile` explicitly ask for the desired permissions on the file being created.